metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MysqlUserMetadataService.java

```diff
@@ -331,16 +345,29 @@ private void _deleteDataMetadatas(
     @Transactional(readOnly = true)
     public Map<String, ObjectNode> getDefinitionMetadataMap(
         @Nonnull final List<QualifiedName> names) {
-        if (!names.isEmpty()) {
-            final List<List<QualifiedName>> parts = Lists.partition(names, config.getUserMetadataMaxInClauseItems());
-            return parts.stream()
+        final List<QualifiedName> oNames = names.stream().filter(name -> !name.isPartitionDefinition()).collect(
```

Comment:
Does this function returns both the table definition metadata map and the partition metadata map? If we don't want to change the API, we should at least separate these two parts as two functions internally for clarity.

Reply:
Yes, it does. Since sharding the table is an internal change for optimization purpose, we do not have a requirement to expose an API to separate them.For clarity,  I will create internal private methods.